### PR TITLE
Merge Kaggle integration to CV `master` branch

### PR DESCRIPTION
**Pull request by nkovela1**
This PR merges the Kaggle integration into the KerasCV master branch.

All presets will now be hosted via Kaggle, and all KerasCV models and presets are now public on Kaggle: https://www.kaggle.com/organizations/keras/models

### CODE REVIEW

File: keras_cv/utils/preset_utils_test.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        input_batch = np.ones(shape=(2, 224, 224, 3))
+        expected_output = model(input_batch)
+        restored_output = restored_model(input_batch)
+        self.assertAllClose(expected_output, restored_output)
+
+    def test_preset_errors(self):
+        with self.assertRaisesRegex(ValueError, "must be a string"):
+            ImageClassifier.from_preset(ImageClassifier)
+        with self.assertRaisesRegex(ValueError, "Unknown preset identifier"):
+            ImageClassifier.from_preset("taggle://rednet/rednet/rednet")
```
Comment (sampathweb) [inline keras_cv/utils/preset_utils_test.py:87]: Should this be `kaggle`?

Comment (sampathweb) [reply keras_cv/utils/preset_utils_test.py:87]: Oh.  This tests failure.  Resolving it.

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_test.py [hunk 2]
```diff
@@ -113,6 +113,9 @@ def test_saved_alias_model(self):
         )
 
     def test_feature_pyramid_inputs(self):
+        import os
+
+        print("HERE", os.environ["KAGGLE_USERNAME"])
```
Comment (sampathweb) [inline keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_test.py:118]: Do you need this print?

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py [hunk 3]
```diff
@@ -26,15 +26,7 @@
...
-        "class_name": "keras_cv>CSPDarkNetBackbone",
-        "config": {
-            "stackwise_channels": [48, 96, 192, 384],
-            "stackwise_depth": [1, 3, 3, 1],
-            "include_rescaling": True,
-            "use_depthwise": False,
-            "input_shape": (None, None, 3),
-            "input_tensor": None,
-        },
+        "kaggle_handle": "kaggle://keras/cspdarknet/csp_darknet_tiny/1",
```
Comment (sampathweb) [inline keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py:29]: NIT: Why name this `kaggle_handle`, when you can also give `gs://` for GCS.  Why not have it as generic name so we can also extend it to future like `hf://`, vertext AI endpoint or model garden?

Comment (nkovela1) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py:29]: This is intended to remain with a Kaggle source of truth. The use of gs:// was temporary for testing while we transition to this, and we intend on hosting all our NLP models on Kaggle.

**Review by divyashreepathihalli — APPROVED** [review]

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py (hunk 3 above)
Comment (mattdangerw) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py:29]: Yeah, extending our hosting options to `gs://`, `s3://`, `hf://` sounds great, but we should keep all of our Keras team models in one place--Kaggle models. Anything else will make for a fractured experience (i.e. with offline use in Kaggle notebooks).

**Pull request merged**